core/org.wso2.carbon.core/src/test/java/org/wso2/carbon/core/util/KeyStoreUtilTest.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.core.util;
+
+import org.testng.annotations.Test;
+import org.wso2.carbon.CarbonException;
+import org.wso2.carbon.utils.ServerConstants;
+
+import javax.xml.namespace.QName;
+
+import static org.testng.Assert.assertThrows;
+import static org.testng.AssertJUnit.*;
+
+class KeyStoreUtilTest {
```

Comment:
Don't we need to add this to the testng.xml file ? And please add tests for reading the custom keystore file as well

Reply:
Fixed in https://github.com/wso2/carbon-kernel/pull/4025/commits/93a9c6449fc040fff285241a696a96ec84ec4589